connd/qconnectionagent.cpp

```diff
@@ -490,25 +515,25 @@ bool QConnectionAgent::isStateOnline(const QString &state)
 
 void QConnectionAgent::techTetheringChanged(bool on)
 {
-    qDebug() << on;
-    tetheringEnabled = on;
+    qCDebug(connAgent) << on;
     NetworkTechnology *technology = static_cast<NetworkTechnology *>(sender());
-
-    if (on && delayedTethering && technology) {
+    if (technology == tetheringBtTech && on) {
```

Comment:
Instead of comparing pointer addresses I'd rather compare to `technology && (technology->type() == QLatin1String("bluetooth") && on)`